### PR TITLE
add BLAST programs

### PR DESCRIPTION
**Pull request by mjmansfi**
# luslab-nf-modules

Thank you for contributing and submitting a pull request!

Please check you have done the following:

## Uniform coding style

- [x] Used snake_case rather than camelCase
- [x] Followed the output channel naming convention (e.g. `bam` for BAM files, `bed` for BED files)

## Continuous integration

- [x] Added any new modules to `.github/workflows/modules-testing.yml`
- [x] Added any new containers to `.github/workflows/docker-build-push.yml` and `.github/workflows/docker-linting.yml`
- [x] Added any new workflows to `.github/workflows/workflow-testing.yml`
- [x] Added any new test data to `test_data`
- [x] When changing an existing module, have you bumped the version? Major.Minor.Patch - Major for breaking changes, Minor for feature updates and most things, Patch for bug fixes and small things like comment and variable name updates

## Notes

Modelled mostly after my LAST, with some differences. Each BLAST search method is a different program, so I had to add a bunch of somewhat redundant processes and entries in `module-defaults.config`. In the future maybe this could just be ${SEARCH_PROGRAM} [etc] somehow.

### CODE REVIEW

**Comment by mjmansfi** [comment]
Guppy is failing because of `ca-certificates`. I will patch, but it will have to wait until Monday.

**Comment by mjmansfi** [comment]
Should be good now.

File: tools/blast/main.nf [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    container "quay.io/biocontainers/blast:2.9.0--pl526he19e7b1_7"
+
+    input:
+        val opts
+        tuple val(meta), path(ref_fasta)
+        tuple val(meta), path(blast_db)
+        tuple val(meta), path(query_fasta)
+
+    output:
+        tuple val(meta), path("*.asn"), emit: blast_output
```
Comment (chris-cheshire) [inline tools/blast/main.nf:65]: can this be changed to file ext (asn)

File: tools/blast/main.nf [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
```
Comment (chris-cheshire) [inline tools/blast/main.nf:106]: Same here

File: tools/blast/main.nf [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
```
Comment (chris-cheshire) [inline tools/blast/main.nf:147]: Same here

File: tools/blast/main.nf [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
```
Comment (chris-cheshire) [inline tools/blast/main.nf:188]: same here

File: tools/blast/main.nf [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
```
Comment (chris-cheshire) [inline tools/blast/main.nf:229]: You get the idea lol

File: tools/blast/test/main.nf [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Channel
+    .from(test_query_sequence)
+    .map { row -> [ row[0], file(row[1], checkIfExists: true) ] }
+    .set {ch_query_sequence}
+
+/*------------------------------------------------------------------------------------*/
+/* Run tests
+--------------------------------------------------------------------------------------*/
+
+workflow {
```
Comment (chris-cheshire) [inline tools/blast/test/main.nf:54]: Does this test all the modules?

**Review by chris-cheshire — CHANGES_REQUESTED** [review]

**Comment by mjmansfi** [comment]
Added quite a few things. 

- fix `blast_output` channel output names
- test all search programs in the test set; required adding new test files
- I had to create a `blast.config` to set various parameters. This was to prevent the nucleotide and protein databases from ever conflicting with one another

**Review by chris-cheshire — APPROVED** [review]

**Pull request merged**